lib/omnibus/packagers/appx.rb

```diff
@@ -0,0 +1,124 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require 'omnibus/packagers/windows_base'
+
+module Omnibus
+  class Packager::APPX < Packager::WindowsBase
+    id :appx
+
+    setup do
+      # Render the manifest
+      write_manifest_file
+
+      # Copy all the staging assets from vendored Omnibus into the resources
+      # directory.
+      FileSyncer.glob("#{Omnibus.source_root}/resources/#{id}/assets/*").each do |file|
+        copy_file(file, "#{project.install_dir}/#{File.basename(file)}")
+      end
+
+      # Copy all assets in the user's project directory - this may overwrite
+      # files copied in the previous step, but that's okay :)
+      FileSyncer.glob("#{resources_path}/assets/*").each do |file|
+        copy_file(file, "#{project.install_dir}/#{File.basename(file)}")
+      end
+    end
+
+    build do
+      # Pack the files with makeappx.exe, recursively generate fragment for
+      # project directory
+      Dir.chdir(staging_dir) do
+        appx_file = windows_safe_path(Config.package_dir, package_name)
+        shellout!(pack_command(appx_file))
+
+        if signing_identity
+          sign_package(appx_file)
+        end
+      end
+    end
+
+    # @see Base#package_name
+    def package_name
+      "#{project.package_name}-#{project.build_version}-#{project.build_iteration}.appx"
+    end
+
+    #
+    # Write the manifest file into the staging directory.
+    #
+    # @return [void]
+    #
+    def write_manifest_file
+      render_template(resource_path('AppxManifest.xml.erb'),
+        destination: "#{windows_safe_path(project.install_dir)}/AppxManifest.xml",
+        variables: {
+          name:            project.package_name,
+          friendly_name:   project.friendly_name,
+          version:         appx_version,
+          maintainer:      project.maintainer,
+          certificate_subject: certificate_subject,
+        }
+      )
+    end
+
+    #
+    # Get the shell command to run pack in order to create a
+    # an appx package
+    #
+    # @return [String]
+    #
+    def pack_command(appx_file)
+      "makeappx.exe pack /d \"#{windows_safe_path(project.install_dir)}\" /p #{appx_file}"
+    end
+
+    #
+    # Parse and return the APPX version from the {Project#build_version}.
+    #
+    # A project's +build_version+ looks something like:
+    #
+    #     dev builds => 11.14.0-alpha.1+20140501194641.git.94.561b564
+    #                => 0.0.0+20140506165802.1
+    #
+    #     rel builds => 11.14.0.alpha.1 || 11.14.0
+    #
+    # The APPX version spec expects a version that looks like X.Y.Z.W where
+    # X, Y, Z & W are all 32 bit integers.
+    #
+    # @return [String]
+    #
+    def appx_version
+      versions = project.build_version.split(/[.+-]/)
+      "#{versions[0]}.#{versions[1]}.#{versions[2]}.#{project.build_iteration}"
```

Comment:
cool. I basically did a direct copy/paste here from the `msi_version` which reminds me since we made a base class, thats the propper home for this and I'll change the implementation to match above